docs/Quality.md

```diff
@@ -103,3 +112,40 @@ This process can happen automatically with [GitLab's CI pipelines](https://docs.
 
 ## Continuous Deployment
 TODO
+
+## Release Process
+A good release process is essential in maintaining quality. It's important to consider:
+- What is going into a release?
+- Has it been tested before releasing?
+- What happens if a previous release needs to be patched?
+- What versions of software components integrate with each other?
+
+### Software Integration
+Software rarely comes in singles. There is usually at least a client side application that pairs with server side code. If done well, the server code should always remain backwards compatible. Meaning, any updates to the server should continue to maintain previous versions of client application that users haven't updated yet.
+
+It's still a good idea to capture and release software in groups. Especially when you have multiple types of client software, such as a dashboard that pairs with a client application, and they both share the same server backend.
+
+The best approach is to add **git tags** with [Semantic Versioning](https://semver.org/) to each of the software components being integrated. The tags will make it easy for developers to roll back to the exact software for that version.
```

Comment:
link to something like https://git-scm.com/book/en/v2/Git-Basics-Tagging to explain what `git tags` are

Reply:
what do you think about creating releases for each major/Named version?
https://help.github.com/articles/creating-releases/